pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -89,20 +90,22 @@ func (c *Cluster) makeDeployment(nodeName string, selection rookalpha.Selection,
 	volumes := opspec.PodVolumes(c.dataDirHostPath, c.Namespace)
 
 	var dataDir string
+	var dataPathMap *opconfig.DataPathMap
 	if osd.IsDirectory {
 		// Mount the path to the directory-based osd
-		// osd.DataPath includes the osd subdirectory, so we want to mount the parent directory
+		// osd.DataPath includes the osd subdirectory, and this is what Rook uses to determine OSDs
 		parentDir := filepath.Dir(osd.DataPath)
 		dataDir = parentDir
-		// Skip the mount if this is the default directory being mounted. Inside the container, the path
-		// will be mounted at "/var/lib/rook" even if the dataDirHostPath is a different path on the host.
-		if parentDir != k8sutil.DataDir {
-			volumeName := k8sutil.PathToVolumeName(parentDir)
-			dataDirSource := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: parentDir}}
-			volumes = append(volumes, v1.Volume{Name: volumeName, VolumeSource: dataDirSource})
-			configVolumeMounts = append(configVolumeMounts, v1.VolumeMount{Name: volumeName, MountPath: parentDir})
-			volumeMounts = append(volumeMounts, v1.VolumeMount{Name: volumeName, MountPath: parentDir})
+		// for directory osds, we completely overwrite the starting point from above.
+		dataPathMap = &opconfig.DataPathMap{
+			HostDataDir:      osd.DataPath,
+			ContainerDataDir: path.Join(opconfig.VarLibCephDir, "osd", fmt.Sprintf("ceph-%d", osd.ID)),
```

Comment:
This just the path inside the container and it will still be in the same place on the host path, correct? Otherwise I guess we should see an issue in the upgrade tests.

Reply:
Yes. The `HostDataDir: osd.DataPath` handles the (same) on-host location.